src/Admin/AdminHelper.php

```diff
@@ -252,6 +230,26 @@ public function appendFormFieldElement(AdminInterface $admin, $subject, $element
             $newInstance = ObjectManipulator::addInstance($form->getData(), $associationAdmin->getNewInstance(), $fieldDescription);
 
             $associationAdmin->setSubject($newInstance);
+        } else {
+            //Else branch is executed when child form not found (probably nested one)
+            //if childFormBuilder was not found resulted in fatal error getName() method call on non object
+            //OR if childFormBuilder was found, but admin object does not have formFieldDescription with
+            //same name as childFormBuilder
+
+            $path = $this->getElementAccessPath($elementId, $subject);
+
+            $collection = $this->propertyAccessor->getValue($subject, $path);
+
+            if ($collection instanceof Collection) {
+                $modelClassName = $this->getEntityClassName($admin, explode('.', preg_replace('#\[\d*?\]#', '', $path)));
+            } else {
+                throw new \Exception('unknown collection class');
```

Comment:
IMO, we could flip this condition in order to avoid the `else` block.
Regarding the exception, I think we must use a narrower type (like `\TypeError` or similar) and provide a better message explaining what is the expected and the found type.
```suggestion
                throw new \Exception('Unknown collection class.');
```

Reply:
Did something that you asked for

Reply:
Thank you!